### DISCUSSION TITLE
Update to 2.4.90 fails with error 1: Unhandled error

### DISCUSSION THREAD

**Discussion by hotcobra** [Security-Onion-Solutions/securityonion#13514] [2.4]
### Version

2.4.90

### Installation Method

Security Onion ISO image

### Description

upgrading

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Meets minimum requirements

### CPU

6

### RAM

16g

### Storage for /

 67G

### Storage for /nsm

130G

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

No, there are no additional clues

### Detail

- Soup failed with error 1: Unhandled error
       Unhandled error occured, please check /root/soup.log for details.
[yum50.log](https://github.com/user-attachments/files/16650917/yum50.log)
[soup.log](https://github.com/user-attachments/files/16650922/soup.log)
[so-status.log](https://github.com/user-attachments/files/16650955/so-status.log)

-    Soup.log reflects can't connect to salt-master but salt-master is running:
           salt-master.service - The Salt Master Server
                  Loaded: loaded (/usr/lib/systemd/system/salt-master.service; enabled; preset: disabled)
                  Active: active (running) since Sat 2024-08-17 16:32:03 UTC; 24h ago
                  Docs: man:salt-master(1)
                        file:///usr/share/doc/salt/html/contents.html
                        https://docs.saltproject.io/en/latest/contents.html
                        Main PID: 956 (/opt/saltstack/)
                       Tasks: 42 (limit: 99170)

- So-status reflects so-elastalert,so-nginx and so-elasticsearch status as "missing".
   
- sudo salt-call state.show_top shows:
       Pillar timed out after 90 seconds after :
       [INFO    ] Caching directory '_grains/' for environment 'base'

- sudo salt-call state.apply salt.minion -l info queue=True 
   shows  same thing:Pillar timed out after 90 seconds

- Checking yum history: On August 2nd, Scriptlet shows:
usr/sbin/weak-modules: line 1089: cd: /lib/modules/5.14.0-427.24.1.el9_4.x86_64/weak-updates: No such file or directory

no /opt/so/saltstack/default/salt/top.sls.backup

### Guidelines

- [X] I have read the discussion guidelines at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720 and assert that I have followed the guidelines.

**Comment by cm-ops** [comment]
Is there anything of note in the master log?

Also check the salt-minion service and minion log `/opt/so/log/salt/minion`.

What do these show? `cat /etc/soversion`  and docker ps.

**Reply by hotcobra** [reply]
Only INFO entries in master log like:
2024-08-19 00:09:51,040 [salt.client.mixins:441 ][INFO    ][3131] Runner completed: 20240819000950708904

salt-minion service is running:
salt-minion.service - The Salt Minion
     Loaded: loaded (/usr/lib/systemd/system/salt-minion.service; enabled; preset: disabled)
     Active: active (running) since Mon 2024-08-19 18:34:38 UTC; 2min 13s ago
       Docs: man:salt-minion(1)
             file:///usr/share/doc/salt/html/contents.html
             https://docs.saltproject.io/en/latest/contents.html
    Process: 3747005 ExecStartPre=/bin/sleep 30 (code=exited, status=0/SUCCESS)
   Main PID: 3748408 (python3.10)
      Tasks: 11 (limit: 99170)
     Memory: 58.5M
        CPU: 12.808s
     CGroup: /system.slice/salt-minion.service
             ├─3748408 /opt/saltstack/salt/bin/python3.10 /usr/bin/salt-minion
             └─3748451 "/opt/saltstack/salt/bin/python3.10 /usr/bin/salt-minion MultiMinionProcessManager MinionProcessManager"
Aug 19 18:34:06 so245020 salt-minion[3734723]:   File "/opt/saltstack/salt/lib/python3.10/site-packages/zmq/eventloop/zmqstream.py", line 628, in _check_closed
Aug 19 18:34:06 so245020 salt-minion[3734723]:     raise OSError("Stream is closed")
Aug 19 18:34:06 so245020 salt-minion[3734723]: OSError: Stream is closed
Aug 19 18:34:38 so245020 systemd[1]: Started The Salt Minion.
Aug 19 18:34:42 so245020 salt-minion[3748451]: [INFO    ] Setting up the Salt Minion "so245020_standalone"
Aug 19 18:34:44 so245020 salt-minion[3748451]: [INFO    ] Starting up the Salt Minion
Aug 19 18:34:44 so245020 salt-minion[3748451]: [INFO    ] Starting pull socket on /var/run/salt/minion/minion_event_2a32bdbc97_pull.ipc
Aug 19 18:34:48 so245020 salt-minion[3748451]: [INFO    ] Creating minion process manager
Aug 19 18:35:48 so245020 salt-minion[3748451]: [ERROR   ] Minion unable to successfully connect to a Salt Master.
Aug 19 18:36:21 so245020 salt-minion[3748451]: [ERROR   ] Error while bringing up minion for multi-master. Is master at so245020 responding?

etc/soversion and all docker ps show 2.4.80 so I'm on 2.4.80 but down. 

salt-minion can't connect to salt-master.

**Comment by hotcobra** [comment]
Think I'll just drop SO and reinstall.

**Reply by cm-ops** [reply]
Okay, was going to see if you ran `sudo salt-key -d so245020_standalone`, then `sudo systemctl restart salt-minion`, and finally `sudo salt-key -A `(and select yes to accept the minion key). Wanted to see if the minion is able to send the new key to the master and have the master accept.

**Comment by hotcobra** [comment]
Sorry, 

I suspect your’e right that it was a key issue.It’s a standalone so I just used this as a good training environment.  



Hopefully a complete install will work….we’ll see


John Mulkerin
***@***.***
